.github/workflows/check_sp.yml

```diff
@@ -0,0 +1,48 @@
+name: Check Service Principal
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "35 6 * * *"
+
+jobs:
+  set-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      tests: ${{ steps.select-tests.outputs.tests }}
+    steps:
+      - name: Select Tests
+        id: select-tests
+        run: |
+          d="{'environment' :'Development'  , 'principal': 's121d01-keyvault-readonly-access'}"
+          t="{'environment' :'Test'         , 'principal': 's121t01-keyvault-readonly-access'}"
+          p="{'environment' :'Production'   , 'principal': 's121p01-keyvault-readonly-access'}"
+          tests="{ 'data':[ ${d} ,  ${t} ,  ${p} ]}"
```

Comment:
As discussed, this can be simplified as a list of service principal strings. Directly in the matrix strategy so we can remove this job